Fiscal Year Pivot Click Through Fix: GS-3431

positiveVibes4all opened the pull request:
## Pivot Chart Intersection Selection Filtering Fix
---
- Fixed fiscal year click through and added hoisting the the filters.
- Refactored code that was impacted.

decrapifier commented:
<!--
  3 failure:  Your PR has lint ..., The changelog has..., The version was n...
  3 warning:  You should add at..., Your PR has lint ..., The README has no...
  1 messages
  
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>index.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/index.js</code></p>
  
  <strong>Async arrow function has no 'await' expression.</strong>
  <code>Line 26: let runTypeProcessor = _.curry(async (getProvider, processor, item, ...args) => {</code>
</details>
<details>
  <summary>index.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/provider-debug/index.js</code></p>
  
  <strong>'_' is assigned a value but never used.</strong>
  <code>Line 1: let _ = require('lodash/fp')</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You should add at least 1 more reviewer to the PR
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Your PR has lint warnings. Please consider fixing these.
      
<details>
  <summary>utils.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/utils.js</code></p>
  
  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 43: if (group.join == 'or' && currentKey)</code>

  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 52: item => item.key == currentKey && !getItems(item),</code>

  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 60: if (relevantFilters.length == 1 && group.join !== 'not')</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

Commit:
FY Clickthrough Fix

- Fixed click through of pivot chart to result table based on cell intersection.
- hoisting will now take a set of properties from an intervals filter to above the query
- Bump version by a minor version based on this being a new feature.

Commit:
Cleanup

- Remove accidental addition to package.json file.
- Fix CHANGELOG Details

daedalus28 reviewed (changes requested):
Based on the feedback in the original PR, the refactoring here should really move hoisting to a new pivot type function for hoisting so we can get rid of the omit juggling.

The diff here also seems very large - we should probably split the bug fix and refactoring into two PRs (and judging from this diff, I think perhaps we should start with the refactor)

daedalus28 commented on packages/provider-elasticsearch/src/example-types/metricGroups/pivot.js:
```diff
@@ -137,10 +146,10 @@ let createPivotScope = (node, schema, getStats) => {
     )(expansions)
 
-  let isFirstOfType = (expansion) =>
+  let isFirstOfType = expansion =>
```
@stellarhoof these seems like prettier diffs - we should have one config for the monorepo and run it for everything in its own PR

stellarhoof replied:
We do, but nothing's running in CI because we need to merge https://github.com/smartprocure/contexture/pull/2 since I deleted my personal token from this repo.

rudolph9 commented on .gitignore:
```diff
@@ -18,3 +18,5 @@ coverage
 !.yarn/sdks
 !.yarn/versions
+
+.vscode/
```
Can you put this in your a local only file to ignore like `.git/info/exclude` or a global gitignore https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

positiveVibes4all replied:
Sure thing

stellarhoof committed:
Merge branch 'master' into feature/GS-3431-Fix-Pivot-Intersection-Filtering

stellarhoof committed:
Fix yarn.lock

Signed-off-by: Alejandro Hernandez <ahernandez@govspend.com>

lint-action committed:
Fix code style issues with Prettier

positiveVibes4all committed:
Merge branch 'master' into feature/GS-3431-Fix-Pivot-Intersection-Filtering

Commit:
Remove Local GitIgnore References

positiveVibes4all commented:
Another opened for this in favor of reducing commits and removing refactor work not related to this commit.
